src/cockpit/transports.py

```diff
@@ -413,6 +413,11 @@ def kill(self) -> None:
 
     def _close(self) -> None:
         if self._process is not None:
+            if self.can_write_eof():
```

Comment:
I think the fix here is wrong (although certainly correct in spirit).

This is more damage from ead84ab7ba1a8a234ca4093d680c5916cea94fa4.

The correct fix is to put something underneath, roughly like:

```python
        if self._process.stdin is not None:
            self._process.stdin.close()
```

at the same point that ead84ab7ba1a8a234ca4093d680c5916cea94fa4 removed the similar code for `self._sock`.